### DISCUSSION TITLE
Implicit datetime snapshots

### DISCUSSION THREAD

**Discussion by ozialien** [openzfs/zfs#15184] [Ideas]
I am wondering what the practicalities would be to add implicit datetime snapshots.  Would this imply a uberblock change journal or something.  I like the idea of being able to go to a datetime and not worry about explicitly snapshoting most of the time.

**Comment by amotin** [comment]
If you create snapshots often and never delete them, it effectively mean that you never free any space.  Plus too many snapshots on a pool with constant write and delete increase time of their deletion exponentially.  Just set up some reasonable snapshot policy you like explicit, it is not difficult.
